### PR TITLE
RATIS-846: create simplest possible example, a replicated counter

### PR DESCRIPTION
**Pull request by esahekmat**

### CODE REVIEW

File: ratis-examples/src/main/java/org/apache/ratis/examples/counter/server/CounterServer.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        .setGroup(CounterCommon.RAFT_GROUP)
+        .setProperties(properties)
+        .setServerId(currentPeer.getId())
+        .setStateMachine(counterStateMachine)
+        .build();
+    server.start();
+
+    //exit when any input entered
+    Scanner scanner = new Scanner(System.in);
+    scanner.nextLine();
```
[the remark this answers is not in the document]
Comment (esahekmat) [reply ratis-examples/src/main/java/org/apache/ratis/examples/counter/server/CounterServer.java:61]: I tried this shutdown hook with no success, so I decided to go this way,
     Runtime.getRuntime().addShutdownHook(new Thread(()-> {
      try {
        server.close();
      } catch (IOException e) {
        e.printStackTrace();
      }
    }));

File: ratis-examples/README.md [hunk 2]
```diff
@@ -114,7 +118,24 @@ Continue the server command example,
...
+`CounterServer` class contains the main method to run the server and you can run it 
+three times with three different parameters(1,2 and 3).
+all address and ports of the peers hardcoded in `CounterCommon`, so you don't 
+need any extra configuration to run this example on your localhost.
+`CounterClient` class contains the main method to run the client,the client sends 
+several INCREMENT command and after that, it sends a GET command and prints the 
+result which should be the value of the counter.
+'Counter State Machine' implemented in `CounterStateMachine` class.
+You can find more detail by reading these classes javaDocs.
+
```
[the remark this answers is not in the document]
Comment (esahekmat) [reply ratis-examples/README.md:134]: I have moved them

File: ratis-examples/README.md [hunk 3]
```diff
@@ -19,9 +19,13 @@
 The repository can be complied using `mvn clean package -DskipTests` under the project root directory;
 see also [BUILDING.md](../BUILDING.md).
 
-All the scripts for running the examples are located in the [ratis-examples/src/main/bin](src/main/bin) directory;
+For the Example 1 and 2, All the scripts for running the examples are located in the [ratis-examples/src/main/bin](src/main/bin) directory;
 see below for the usage.
 
+Example 3 does not contain any script to run it thus you can run it by these commands:
+for server: `java -cp target/*.jar org.apache.ratis.examples.counter.server.CounterServer`
```
[the remark this answers is not in the document]
Comment (esahekmat) [reply ratis-examples/README.md]: I have moved this into run section

**Comment by jnp** [comment]
Thanks for addressing comments. I am +1 for the patch.

**Comment by esahekmat** [comment]
Any other comment or opionion?

**Comment by esahekmat** [comment]
I have improved the PR and addressed all the * imports, current failure in build(unit tests) is not related to my work

**Pull request merged**